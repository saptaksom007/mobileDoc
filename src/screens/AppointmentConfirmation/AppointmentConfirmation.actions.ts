export const ActionTypes = {
  APPOINTMENT_CONFIRMATION_REQUESTED: 'APPOINTMENT_CONFIRMATION_REQUESTED',
  APPOINTMENT_CONFIRMATION_SUCCEEDED: 'APPOINTMENT_CONFIRMATION_SUCCEEDED',
  APPOINTMENT_CONFIRMATION_FAILED: 'APPOINTMENT_CONFIRMATION_FAILED',
  MARK_AS_READ_APPOINTMENT_CONFIRMATION: 'MARK_AS_READ_APPOINTMENT_CONFIRMATION'
}

export default class Actions {
  static appointmentConfirmationRequested(id: number) {
    return {
      type: ActionTypes.APPOINTMENT_CONFIRMATION_REQUESTED,
      payload: id
    }
  }
  static appointmentConfirmationSucceeded(data: any) {
    return {
      type: ActionTypes.APPOINTMENT_CONFIRMATION_SUCCEEDED,
      payload: data
    }
  }
  static appointmentConfirmationFailed(error: Error) {
    return {
      type: ActionTypes.APPOINTMENT_CONFIRMATION_FAILED,
      payload: error
    }
  }
  static markAsReadAppointmentConfirmation(id: number) {
    return {
      type: ActionTypes.MARK_AS_READ_APPOINTMENT_CONFIRMATION,
      payload: id
    }
  }
}
