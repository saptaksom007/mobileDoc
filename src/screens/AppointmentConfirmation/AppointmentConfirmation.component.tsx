import React, { Component } from 'react'
import { compose } from 'recompose'
import { propOr } from 'ramda'
import { View, ScrollView } from 'react-native'
import { connect } from 'react-redux'
import { Card, ListItem } from 'react-native-elements'
import i18n from 'ex-react-native-i18n'
import toString from 'common-docdok/lib/utils/toString'
import {
  momentWithLocales as moment,
  getCalendarOutputFormatWithToday
} from 'utilities/date'

import MessageMyDoctor from 'components/MessageMyDoctor/MessageMyDoctor.component'
import { Avatar } from 'components/Avatar/Avatar.component'
import ExportAppointmentLink from 'components/ExportAppointmentLink/ExportAppointmentLink.component'
import Clinic from 'components/Clinic/Clinic.component'
import { fontFamily } from 'components/CustomText/CustomText.styles'
import {
  Paragraph,
  Bold,
  Medium
} from 'components/CustomText/CustomText.component'

import withSubtitle from 'screens/enhancers/withSubtitle'
import Actions from './AppointmentConfirmation.actions'
import styles from './AppointmentConfirmation.styles'
import { Action } from 'redux'
import { NavigationScreenProp } from 'react-navigation'

interface Props {
  dispatch(action: Action): void
  navigation: NavigationScreenProp<any>
  name: string
  appointment: any & { physicianAvatarPicture: string }
  conversationId: number
}

function getSafeFromAppointment(key: string, appointment: any) {
  return propOr<string, any, string>(
    i18n.t('global.undefined'),
    key,
    appointment
  )
}

function getAppointmentTime(appointment: any) {
  return moment(appointment.appointmentTime || '2017-06-19').locale(
    i18n.getFallbackLocale()
  )
}

export class AppointmentConfirmation extends Component<Props> {
  componentDidMount() {
    const {
      navigation: {
        state: { params }
      },
      dispatch
    } = this.props

    dispatch(
      Actions.appointmentConfirmationRequested(params && params.appointmentId)
    )
    dispatch(
      Actions.markAsReadAppointmentConfirmation(params && params.appointmentId)
    )
  }

  renderDoctorAndSchedule() {
    const { appointment } = this.props
    const appointmentTime = getAppointmentTime(appointment)
    const physician = getSafeFromAppointment('physician', appointment)
    return (
      <Medium style={styles.message}>
        <Bold>{i18n.t('appointment.card.doctor')}:</Bold> {physician}
        {'\n'}
        <Bold>{i18n.t('appointment.card.date')}:</Bold>{' '}
        {appointmentTime.format('L')}
        {'\n'}
        <Bold>{i18n.t('appointment.card.time')}:</Bold>{' '}
        {appointmentTime.format('LT')}
      </Medium>
    )
  }

  renderClinic() {
    const { appointment } = this.props
    const clinic = getSafeFromAppointment('clinicName', appointment)
    const street = getSafeFromAppointment('clinicStreet', appointment)
    const location = getSafeFromAppointment('clinicLocation', appointment)
    const phoneNumber = getSafeFromAppointment('clinicContact', appointment)
    return (
      <Clinic name={clinic} contact={phoneNumber} {...{ street, location }} />
    )
  }

  render() {
    const { name, appointment, conversationId } = this.props

    if (!appointment) {
      return null
    }

    const appointmentTime = getAppointmentTime(appointment)
    const avatar = (
      <Avatar
        name={appointment.physician}
        url={appointment.physicianAvatarPicture}
      />
    )

    return (
      <ScrollView contentContainerStyle={styles.contentContainerStyle}>
        <View style={styles.messageContainer}>
          <Paragraph style={styles.message}>
            <Paragraph>
              {i18n.t('appointment.salutation', {
                name,
                default: 'Hello,'
              })}
            </Paragraph>
            {'\n'}
            {'\n'}
            <Paragraph>
              {i18n.t('appointmentConfirmation.message', {
                date: appointmentTime.format('Do'),
                day: appointmentTime.format('dddd'),
                hour: appointmentTime.format('LT'),
                month: appointmentTime.format('MMMM'),
                calendar: appointmentTime.calendar(
                  undefined,
                  getCalendarOutputFormatWithToday()
                ),
                physician: appointment.physician
              })}
            </Paragraph>
          </Paragraph>
        </View>
        <Card
          containerStyle={styles.cardContainerStyle}
          title={i18n.t('appointment.card.title')}
          titleStyle={styles.cardTitle}
          fontFamily={fontFamily}
        >
          <ListItem
            avatar={avatar}
            roundAvatar
            titleNumberOfLines={3}
            title={this.renderDoctorAndSchedule()}
            titleContainerStyle={styles.cardFirstItemTitle}
            hideChevron
            containerStyle={styles.cardLastItem}
          />

          <ListItem
            title={this.renderClinic()}
            titleContainerStyle={styles.cardFirstItemTitle}
            hideChevron
            containerStyle={[
              styles.cardLastItem,
              !appointment.message && styles.cardLastItemNoBorder
            ]}
          />
        </Card>
        <View style={styles.bottomLinks}>
          <MessageMyDoctor
            containerStyle={styles.messageMyDoctor}
            forceConversationId={conversationId}
          />
          <ExportAppointmentLink appointment={appointment} />
        </View>
      </ScrollView>
    )
  }
}

function select({ profile, appointmentconfirmation }: any) {
  const name = (profile && ` ${toString.person(profile)}`) || ''
  return { appointment: appointmentconfirmation, name }
}

export default compose<Props, any>(
  connect(select),
  withSubtitle('appointmentConfirmation.subtitle')
)(AppointmentConfirmation)
