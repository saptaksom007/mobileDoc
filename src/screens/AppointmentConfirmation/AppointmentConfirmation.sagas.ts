import { call, put, takeEvery } from 'redux-saga/effects'

import Actions, { ActionTypes } from './AppointmentConfirmation.actions'
import i18n from 'ex-react-native-i18n'
import { Actions as NavAction } from 'navigation/SagaNavigation'
import { fetchAppointmentByIdAsync } from 'api/appointments'

function* fetchAppointmentConfirmation({ payload: id }: any) {
  try {
    const appointment = yield call(fetchAppointmentByIdAsync, { id })
    yield put(Actions.appointmentConfirmationSucceeded(appointment))
  } catch (err) {
    yield put(Actions.appointmentConfirmationFailed(err))
    yield put(
      NavAction.showLocalError(i18n.t('appointment.errors.fetchAppointment')),
    )
  }
}

export default [
  takeEvery(
    ActionTypes.APPOINTMENT_CONFIRMATION_REQUESTED,
    fetchAppointmentConfirmation,
  ),
]
