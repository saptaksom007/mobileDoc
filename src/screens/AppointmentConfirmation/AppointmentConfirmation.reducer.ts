import { ActionTypes } from './AppointmentConfirmation.actions'

const {
  APPOINTMENT_CONFIRMATION_REQUESTED,
  APPOINTMENT_CONFIRMATION_SUCCEEDED,
  APPOINTMENT_CONFIRMATION_FAILED,
} = ActionTypes

interface State {
  loading?: boolean
}

type Action = {
  type: keyof typeof ActionTypes
  payload: {}
}

const initialState: State = {}

export default function appointmentconfirmation(
  state: State = initialState,
  action: Action,
) {
  switch (action.type) {
    case APPOINTMENT_CONFIRMATION_REQUESTED: {
      return { ...state }
    }

    case APPOINTMENT_CONFIRMATION_SUCCEEDED: {
      return { ...state, ...action.payload }
    }

    case APPOINTMENT_CONFIRMATION_FAILED: {
      return { ...state }
    }

    default:
      return state
  }
}
