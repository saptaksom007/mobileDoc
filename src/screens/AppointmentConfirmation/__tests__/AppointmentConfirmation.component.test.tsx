import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { AppointmentConfirmation } from '../AppointmentConfirmation.component'
import { NavigationScreenProp } from 'react-navigation'

jest.mock(
  'components/MessageMyDoctor/MessageMyDoctor.component',
  () => 'MessageMyDoctor',
)
jest.mock(
  'components/ExportAppointmentLink/ExportAppointmentLink.component',
  () => 'ExportAppointmentLink',
)

it('<AppointmentConfirmation/> render correctly', () => {
  const tree = renderer
    .create(
      <AppointmentConfirmation
        conversationId={1}
        dispatch={() => {}}
        appointment={{}}
        name=''
        navigation={{ state: { params: {} } } as NavigationScreenProp<any>}
      />,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
