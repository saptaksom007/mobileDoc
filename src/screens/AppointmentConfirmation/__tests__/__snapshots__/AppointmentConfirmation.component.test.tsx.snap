// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppointmentConfirmation/> render correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "height": 1334,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#fff",
          "justifyContent": "flex-start",
          "margin": 15,
          "marginTop": 25,
        }
      }
    >
      <Text
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "TitilliumWeb-Regular",
              "fontSize": 17.5,
              "lineHeight": 22,
              "overflow": "visible",
            },
            Object {
              "color": "rgba(27, 106, 158, 0.85)",
              "fontSize": 17.5,
              "lineHeight": 21,
            },
            Object {
              "textAlign": "left",
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "TitilliumWeb-Regular",
                "fontSize": 17.5,
                "lineHeight": 22,
                "overflow": "visible",
              },
              Object {
                "color": "rgba(27, 106, 158, 0.85)",
                "fontSize": 17.5,
                "lineHeight": 21,
              },
              undefined,
            ]
          }
        >
          appointment.salutation
        </Text>
        

        

        <Text
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "TitilliumWeb-Regular",
                "fontSize": 17.5,
                "lineHeight": 22,
                "overflow": "visible",
              },
              Object {
                "color": "rgba(27, 106, 158, 0.85)",
                "fontSize": 17.5,
                "lineHeight": 21,
              },
              undefined,
            ]
          }
        >
          appointmentConfirmation.message
        </Text>
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 30,
          "width": 750,
        }
      }
    >
      <MessageMyDoctor
        containerStyle={
          Object {
            "height": undefined,
            "marginTop": 0,
            "width": undefined,
          }
        }
        forceConversationId={1}
      />
      <ExportAppointmentLink
        appointment={Object {}}
      />
    </View>
  </View>
</RCTScrollView>
`;
